ThunderCats/UIColor+TCAdditions.m

```diff
@@ -28,12 +28,19 @@
 
 
 #import "UIColor+TCAdditions.h"
+#import "TCInvalidArgument.h"
 
 @implementation UIColor (TCAdditions)
 
 + (UIColor *)tc_colorWithHexString:(NSString *)hexString
 {
     NSString *cleanString = [hexString stringByReplacingOccurrencesOfString:@"#" withString:@""];
+    
+    if ([cleanString length] != 3 && [cleanString length] != 6)
+    {
+        [TCInvalidArgument raiseWithReason:@"invalid hex string"];
+    }
+    
```

Comment:
I'd recommend something like checking if the first character is `#`, and if so, strip off just that character.  After that, make sure the remaining characters are hex values.  Alternatively, a regex might work here.